snmptrap.php

```diff
@@ -22,24 +24,25 @@
     echo "DEBUG!\n";
 }
 
-$entry = explode(',', $argv[1]);
+// Creates an array with trap info
+while ($f = fgets(STDIN)) {
+    $entry[] = $f;
+}
 
-logfile($argv[1]);
+$hostname = trim($entry[0]);
+$ip = str_replace(array("UDP:","[","]"), "", $entry[1]);
+$ip = trim(strstr($ip, ":", true));
 
-// print_r($entry);
-$device = @dbFetchRow('SELECT * FROM devices WHERE `hostname` = ?', array($entry['0']));
+$device = @dbFetchRow('SELECT * FROM devices WHERE `hostname` = ?', [$hostname]);
```

Comment:
You should check the ip field too.  This might work.
```php
$device = dbFetchRow('SELECT * FROM devices WHERE `hostname`=? OR `hostname`=? OR `ip`=?', [$hostname, $ip, inet_pton($ip)]);
```

Remove the @ from these queries